android/app/src/main/java/app/organicmaps/util/Config.java

```diff
@@ -375,19 +376,37 @@ public static boolean isFirstLaunch(@NonNull Context context)
   public static void setFirstStartDialogSeen(@NonNull Context context)
   {
     MwmApplication.prefs(context)
-        .edit()
-        .putBoolean(KEY_MISC_FIRST_START_DIALOG_SEEN, true)
-        .apply();
+                  .edit()
+                  .putBoolean(KEY_MISC_FIRST_START_DIALOG_SEEN, true)
+                  .apply();
+  }
+
+  public static boolean isSearchHistoryEnabled()
+  {
+    return getBool(KEY_PREF_SEARCH_HISTORY, true);
+  }
+
+  public static void setSearchHistoryEnabled(boolean enabled)
+  {
+    setBool(KEY_PREF_SEARCH_HISTORY, enabled);
   }
 
   private static native boolean nativeHasConfigValue(String name);
+
   private static native boolean nativeDeleteConfigValue(String name);
+
```

Comment:
Please remove all unnecessary formatting changes and keep only relevant logic changes in the commit.